libs/agno/agno/os/app.py

```diff
@@ -334,6 +344,36 @@ def _add_built_in_routes(self, app: FastAPI) -> None:
             self.interfaces.append(a2a_interface)
             self._add_router(app, a2a_interface.get_router())
 
+    def _check_duplicate_ids(
+        self,
+        entities: Optional[List[Any]],
+    ) -> None:
+        """Check for duplicate IDs in a list of entities.
+
+        Args:
+            entities: List of agents, teams, or workflows
+
+        Raises:
+            ValueError: If duplicate IDs are found
+        """
+        if not entities:
+            return
+
+        seen_ids: List[str] = []
+
+        for entity in entities:
+            entity_id = entity.id
+            # If id is None, generate it the same way initialization does
+            if entity_id is None:
+                if hasattr(entity, "name") and entity.name:
+                    entity_id = generate_id_from_name(entity.name)
+                else:
+                    continue  # Will get unique UUID during initialization
+
+            if entity_id in seen_ids:
```

Comment:
Rather raise for all the duplicates at the end. Otherwise the user will get an error, then fix it, then another error on the next try, then fix it, etc.